test/test_mkldnn.py

```diff
@@ -1469,24 +1469,33 @@ def _lstm_params_list(self):
         params_list = list(params_dict.values())
         return params_list
 
-    def _cast_dtype(self, input, bf16):
+    def _cast_dtype(self, input, bf16, fp16):
         if bf16:
             input = input.to(torch.bfloat16)
+        elif fp16:
+            input = input.to(torch.half)
```

Comment:
Suggest to update to `_cast_dtype(self, input, dtype):`, specify dtype to bfloat16 and float16.

Reply:
Originally we would unconditionally cast to bf16 when specified but now it is possible to bypass, sounds not right to me. Should assert on the support in mkldnn here?

Reply:
Originally `_cast_dtype`  cast input to bf16 according to the flag `bf16` which is set by `dtype == torch.bfloat16 and torch.ops.mkldnn._is_mkldnn_bf16_supported()`. Now we add a option of fp16, so we want to use dtype to check if cast input to bf16/fp16.

Reply:
What is the behavior now if mkldnn doesn't support bf16/fp16? Is there a behavior change from the original code? I guess my major concern is that we pass through things silently.

Reply:
Does the caller guarantee the `_is_mkldnn_bf16_supported()` returns `True` when the dtype is bfloat16 (same for fp16)? If so, should we use assertion here on `_is_mkldnn_bf16_supported()`?

Reply:
When initially supporting BF16 LSTM, bf16 check was overlooked Added a check.
Revised `test_lstm`. Removed `_is_mkldnn_bf16_supported`.